### DISCUSSION TITLE
What would you love to see in embedchain?

### DISCUSSION THREAD

**Discussion by taranjeet** [mem0ai/mem0#1080] [General]
Starting this discussion to collect requests from the community.
Please share what would you like to see in embedchain and how can we make it better for you.

**Comment by vineeth-balusani** [comment]
I am not sure if Google's models can be integrated with embedchain or not but it would be great if we could use Gemini and Palm2 models

**Comment by pratikkotian04** [comment]
I would like to see the ability to add multiple files and data at once using embedchain

**Comment by br-follow** [comment]
Structured output defined via pydantic models. Like Marvin or Instructor.

**Comment by DrDavidL** [comment]
Parallel creation of a knowledge graph for context retrieval and response processing.

**Comment by DrDavidL** [comment]
Consider adding settings to control each of these steps in this recent article: https://www.anthropic.com/news/contextual-retrieval while taking advantage of the Claude caching!

**Comment by kinthaiofficial** [comment]
A few things I'd love to see in mem0 from a multi-agent perspective:

**Per-agent memory namespaces** — right now mem0 works well for single-agent/single-user scenarios. For multi-agent systems where many agents share a platform but each needs isolated memory, you want `memory.add(content, agent_id="agent_xyz", user_id="user_abc")` to create true isolation, not just filtered search.

**Time-decay on relevance scoring** — `relevance = importance × (0.95 ^ days_since_stored)` so stale memories naturally fall behind fresher ones in retrieval, without manual pruning. This is how human memory works and it prevents the system from surfacing confidently outdated information.

**Memory consolidation triggers** — a scheduled "dreaming" operation that runs deduction (what can we infer from these memories?) and induction (what pattern applies across these memories?) to promote implicit knowledge to explicit facts. This is different from retrieval; it's an active knowledge synthesis step.

**Cross-agent memory sharing with consent** — when Agent A teaches Agent B a skill, B should be able to store "learned from A" in its memory with a provenance tag. A's knowledge enriches B without giving B full access to A's private memory.

**Memory budget** — each agent has a max memory size (in vectors or storage). When it fills up, the least important + most stale memories are pruned. Makes memory management predictable for operators.

We built a version of these features for KinthAI's agent memory layer — the architecture and design decisions: https://blog.kinthai.ai/why-character-ai-forgets-you-persistent-memory-architecture

Are you planning to extend mem0 beyond single-agent single-user use cases?
